src/main/scala/com/cloudera/sparkts/TimeSeries.scala

```diff
@@ -169,6 +169,55 @@ class TimeSeries[K](val index: DateTimeIndex, val data: DenseMatrix,
   }
 
   /**
+    * Returns a TimeSeries where each individual time series is differenced with the given time
+    * frequency.
```

Comment:
I will add an example to the comments about the function. To answer your specific example:

t            value
1pm      3.5
2pm      3.6
10pm    4.6
11pm    5.9

then the result of differencesByFrequency(new HourFrequency(10)) will be:

t           d(value)
11pm   2.4

The process is as follows: for each timestamp t, find the value at (t - frequency). If there is none, go back to the previous available value. If there is none, discard t because we cannot difference it. This results in discarding all values earlier than <frequency> time after the first available value in the time series, because they cannot be differenced due to missing information. They also cannot be interpolated due to lack of previous values, which is why I don't produce NaNs, instead I truncate the time series.